File: DynamicProgramming1/FibonacciFunction/CPPFibonacciFunction/FibonacciFunction.cpp
#include <iostream>
#include <vector>

#define MAX 41

using namespace std;

vector<int> memo_0(MAX);
vector<int> memo_1(MAX);

void fibonacci_function()
{
	memo_0.push_back(1);
	memo_1.push_back(0);
	memo_0.push_back(0);
	memo_1.push_back(1);

	for (int i = 2; i < MAX; i++)
	{
		memo_0.push_back(memo_0[i - 1] + memo_0[i - 2]);
		memo_1.push_back(memo_1[i - 1] + memo_1[i - 2]);
	}
}

int main()
{
	int t, n;

	fibonacci_function();

	cin >> t;

	for (int i = 0; i < t; i++)
	{
		cin >> n;
		cout << memo_0[n] << " " << memo_1[n] << "\n";
	}

	return 0;
}

File: DynamicProgramming1/FibonacciFunction/JavaFibonacciFunction/src/FibonacciFunction.java
import java.io.*;

public class FibonacciFunction {
    static final int MAX = 41;

    static int[] memo0 = new int[MAX];
    static int[] memo1 = new int[MAX];

    static void fibonacciFunction() {
        memo0[0] = 1;
        memo1[0] = 0;
        memo0[1] = 0;
        memo1[1] = 1;

        for (int i = 2; i < MAX; i++) {
            memo0[i] = memo0[i - 1] + memo0[i - 2];
            memo1[i] = memo1[i - 1] + memo1[i - 2];
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        fibonacciFunction();

        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            int n = Integer.parseInt(br.readLine());
            bw.write(memo0[n] + " " + memo1[n] + "\n");
        }

        bw.flush();
    }
}
